src/index.js

```diff
@@ -140,14 +142,24 @@ class App extends React.Component {
 
   hireAgent(agent) {
     console.log("Agent " + agent.name + " selected");
-    
-    this.setState({
-      selectedAgent: agent,
-      serviceCallComponent: this.serviceNameToComponent[agent.name] || this.serviceDefaultComponent,
-      usingDefaultCallComponent: !(agent.name in this.serviceNameToComponent),
-    });
+    Promise.all([
+      window.fetch(`${agent.endpoint}/encoding`),
+      window.fetch(`http://protobufjs.singularitynet.io/${agent.address}`)
```

Comment:
I've tried with the service `SNET_BR/named-entity-recognition` and I've got:
`GET http://protobufjs.singularitynet.io/0x06b02f360e35f13781704a598ab85f32137c429b 400 (Bad Request)`

Reply:
So, the error message you get associated with the 400 (Bad Request) is "Service spec has more than 1 service".

Here's the big picture:
.proto definitions are compiled by a server component called [service-spec-provider](https://github.com/singnet/service-spec-provider) to a .json format that protobufjs can use to encode/decode the proto requests/responses in the browser, and to provide the dapp with the package, service and method names.

The issue with .proto definitions having more than one service in them, for example sharing the same file or files for multiple services, is identifying which service is actually exposed at the endpoint specified in the Agent contract, either by the application or the user.

You could have one .proto file with 5 services, each of which is deployed at a different endpoint in different servers (or at different ports). So, the user would either see an select input, generated from the protobufjs .json, with the different services available in the source .proto file or files, or have to manually put in the name of a service to call, with no way to know from either the contract or the .json file itself if the services specified in the .proto definitions are actually available at the endpoint specified in the Agent contract.

I can think of two ways to go about this: either we agree on a convention to allow users to specify the name of the service (or services) offered registered with the specified endpoint in the smart contract itself and we communicate that to the users, or we allow developers to only have one service in each .proto definition. I would go with this second option. See [this comment](https://github.com/singnet/alpha-dapp/pull/33#issuecomment-431080179) for the original discussion.